js/framework/historyStore.js

```diff
@@ -319,12 +319,16 @@ var HistoryStore = new Lang.Class({
                     model: model,
                     context_label: model.title,
                 });
-            } else if (model instanceof DModel.Media) {
+            } else if (model instanceof DModel.Media || model instanceof DModel.Audio) {
                 let old_item = this.get_current_item();
+                let context = old_item.model ? old_item.model.resources : null;
+                if (!context || context.length === 0) {
+                    context = [model];
```

Comment:
What does this case cover? There's a similar method, *load_dbus_item*, implemented here as well as in *courseHistoryStore*. In *courseHistoryStore* it does this same thing, setting *context* to the selected model (since it doesn't have an old model to work with).

Reply:
Relatedly, I suspect you'll want to change that other function to include DModel.Audio:

https://github.com/endlessm/eos-knowledge-lib/blob/5743a58f434e0642efc55309d8d10773dcc9826d/js/framework/courseHistoryStore.js#L91

And that's so strange DModel.Audio isn't a DModel.Media subclass :thinking: